### ISSUE TITLE
mlflow.log_params doesn't stringify values, raises TypeError

### ISSUE THREAD

**Issue by tkocmathla** [mlflow/mlflow#1260]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS Mojave
- **MLflow installed from (source or binary)**: binary via pip
- **MLflow version (run ``mlflow --version``)**: mlflow, version 0.9.1
- **Python version**: Python 3.7.2
- **npm version (if running the dev UI)**: n/a
- **Exact command to reproduce**:
```
import mlflow

with mlflow.start_run():
    mlflow.log_params({'foo': 100})
```

### Describe the problem
The docs for mlflow.log_params say that non-string values in the param dictionary will be stringified, but that doesn't happen and a TypeError is raised instead.

### Source code / logs
I think the fix may be as simple as changing https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/fluent.py#L210 from this

     params_arr = [Param(key, value) for key, value in params.items()]

to this

     params_arr = [Param(key, str(value)) for key, value in params.items()]

since that's exactly what the implementation of log_param does:

https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/client.py#L171

**Comment by johny-c** [comment]
This would prevent you from being able to read the types of the parameters when loading a run from the store.

**Comment by tkocmathla** [comment]
Well, there are two problems: the docstring for log_params says vals will be stringified, and the behavior of log_param (singular) is to stringify vals without raising an error.

**Comment by mateiz** [comment]
This does seem like a bug -- thanks for reporting it.

**Comment by smurching** [comment]
Closing this as it's been fixed in MLflow 1.0 via https://github.com/mlflow/mlflow/pull/1364. Please reopen if it's still an issue!
